pkg/crcontrollers/drive/drivecontroller.go

```diff
@@ -311,15 +316,33 @@ func (c *Controller) getVolsStatuses(volumes []*volumecrd.Volume) map[string]str
 	return statuses
 }
 
-func (c *Controller) checkAllVolsRemoved(volumes []*volumecrd.Volume) bool {
-	for _, vol := range volumes {
-		if vol.Spec.CSIStatus != apiV1.Removed {
+// checkAllVolsWithoutFakeAttachRemoved checks if all volumes are removed and not 'fake attached',
+// returns true if any volume's CSIStatus is not 'REMOVED' or it's 'fake attached'
+func (c *Controller) checkAllVolsWithoutFakeAttachRemoved(volumes []*volumecrd.Volume) bool {
+	for _, v := range volumes {
+		if v.Spec.CSIStatus != apiV1.Removed && !c.isFakeAttach(v) {
 			return false
 		}
 	}
 	return true
 }
 
+// checkLVGVolumeFakeAttach checks if there is any volume in the provided LVG
+// that is located on LVM and is fake attached, returns true if such a volume is found, false otherwise.
+func (c *Controller) checkLVGVolumeFakeAttach(lvg *lvgcrd.LogicalVolumeGroup, volumes []*volumecrd.Volume) bool {
+	for _, v := range volumes {
+		if v.Spec.LocationType == apiV1.LocationTypeLVM && v.Spec.Location == lvg.Name {
+			return c.isFakeAttach(v)
```

Comment:
Would it be possible that this function returns false in the following scenario
volumes: [vol1 - fakeattach false, vol2: fakeattach true]